lib/localenv/clusterenv.go

```diff
@@ -107,12 +112,43 @@ func newClusterEnvironment(config clusterEnvironmentConfig) (*ClusterEnvironment
 		return nil, trace.Wrap(err, "failed to connect to etcd")
 	}
 
+	cluster, err := backend.GetLocalSite(defaults.SystemAccountID)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
 	packagesDir, err := SitePackagesDir()
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	objects, err := fs.New(packagesDir)
+	user, err := systeminfo.FromOSUser(cluster.ServiceUser)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	localObjects, err := fs.New(fs.Config{
+		Path: packagesDir,
+		User: user,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	server, err := ops.FindLocalServer(cluster.ClusterState)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	objects, err := libcluster.New(libcluster.Config{
+		Local:       localObjects,
+		WriteFactor: 1,
```

Comment:
Is this going to work with multi-node clusters? The blob will be available on 1 node and when other gravity-sites are coming up they won't be able to find it. Will they be able to replicate it?

Reply:
As far as understand, this is how the active cluster controller works but without the replication loop. As I see this environment's blob service working - it needs to write the package file and update the backend accordingly - as if this was a single node cluster.
Then, the cluster controller should pick it up and distribute it.
I'll have to test this though.